### PR TITLE
Update setup-systemd-resolved

### PR DESCRIPTION
**Pull request by reedlaw**
Select yum or apt-get. Addresses #203

### CODE REVIEW

File: modules/setup-systemd-resolved/setup-systemd-resolved [hunk 1]
```diff
@@ -66,14 +66,30 @@ function assert_not_empty {
...
-  echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
-  sudo apt-get install -y iptables-persistent
+  if has_apt_get; then
+    sudo apt-get update -y
+    echo iptables-persistent iptables-persistent/autosave_v4 boolean true | sudo debconf-set-selections
+    echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
+    sudo apt-get install -y iptables-persistent
+  elif has_yum; then
+    sudo yum update -y
+    sudo yum install -y iptables-services
```
Comment (brikis98) [inline modules/setup-systemd-resolved/setup-systemd-resolved:88]: Do we not need toe `iptables-persistent` calls in this case?

**Review by brikis98 — COMMENTED** [review]
Thanks for the PR!

File: modules/setup-systemd-resolved/setup-systemd-resolved (hunk 1 above)
Comment (reedlaw) [reply modules/setup-systemd-resolved/setup-systemd-resolved:88]: I believe the `iptables-services` package is the Amazon Linux equivalent.

Comment (brikis98) [reply modules/setup-systemd-resolved/setup-systemd-resolved:88]: Does merely _installing_ `iptables-services` do the same thing as the `iptables-persistent` code above though?

Comment (reedlaw) [reply modules/setup-systemd-resolved/setup-systemd-resolved:88]: No, there is further customization needed to `configure_systemd_resolved` below. I am not sure about some of the changes need to `SYSTEMD_RESVOLDED_CONFIG_FILE` (sic). This is needed for Amazon Linux:
```
sudo service iptables save
```

**Comment by brikis98** [comment]
OK, sorry for the long delay. This LGTM. Could you pull in the latest from `master`, and then I can kick off tests?

**Comment by tonyhb** [comment]
@brikis98 rebased here: https://github.com/hashicorp/terraform-aws-consul/pull/228

**Comment by brikis98** [comment]
Thanks @tonyhb. Closing this one.

**Pull request closed without merging**